src/test/java/plugins/MSBuildPluginTest.java

```diff
@@ -0,0 +1,96 @@
+package plugins;
+
+import static org.junit.Assert.assertThat;
+
+import org.hamcrest.Matchers;
+import org.jenkinsci.test.acceptance.junit.AbstractJUnitTest;
+import org.jenkinsci.test.acceptance.junit.Native;
+import org.jenkinsci.test.acceptance.junit.WithOS;
+import org.jenkinsci.test.acceptance.junit.WithPlugins;
+import org.jenkinsci.test.acceptance.msbuild.MSBuildStep;
+import org.jenkinsci.test.acceptance.po.Build;
+import org.jenkinsci.test.acceptance.po.FreeStyleJob;
+import org.junit.Test;
+
+/**
+ * This test is designed to be run on a Windows machine with MSBuild installed and in the PATH
+ * 
+ * Tests will be skipped if:
+ *      - MSBUild.exe is not in the path. This implies that machine running Jenkins is windows.
+ *      - The test and Jenkins are being run on different machines
+ */
+@WithPlugins("msbuild")
+@Native({"MSBuild"})
+@WithOS(os = {WithOS.OS.WINDOWS})
+public class MSBuildPluginTest extends AbstractJUnitTest {
+
+    /**
+     * Builds a FreeStyle job with a MSBuild step with the configuration passed as parameter
+     * 
+     * @param workspacePath The workspace where the project is located
+     * @param buildFile The build file (.proj or .sln)
+     * @param cmdArguments the command line arguments to execute. Can be null.
+     * @return a job
+     * @throws IllegalArgumentException if workspacePath or buildFile are null
+     */
+    private FreeStyleJob msBuildJob(String workspacePath, String buildFile, String cmdArguments) {
+        if (workspacePath != null && buildFile != null) {
+            FreeStyleJob job = jenkins.jobs.create(FreeStyleJob.class);
+            job.copyDir(resource(workspacePath));
+            MSBuildStep msBuildStep = job.addBuildStep(MSBuildStep.class);
+            msBuildStep.setMSBuildFile(buildFile);
+            if (cmdArguments != null && !cmdArguments.isEmpty()) {
+                msBuildStep.setCmdLineArgs(cmdArguments);
+            }
+            job.save();
+            return job;
+        } else {
+            throw new IllegalArgumentException("Workspace and buildFile must be different from null.");
+        }
+    }
+    
+    @Test
+    public void buildProjNoCmdLineArgumentsTest() {
+        FreeStyleJob job = msBuildJob("/msbuild_plugin/projProject/", "project.proj", null);
+
+        // Job should run successfully
+        Build b = job.scheduleBuild();
+        b.shouldSucceed();
+        assertThat(b.getConsole(), Matchers.containsString("Build succeeded."));
+        assertThat(b.getConsole(), Matchers.not(Matchers.containsString("Done building project \"project.proj\".")));
```

Comment:
it's generally better to import static on Hamcrest matchers so it reads better  `not(containsString("foobar)))`

Is this actually the right test - running msbuild without arguments [will search](https://msdn.microsoft.com/en-us/library/ms164311.aspx) for the .proj to run in the CWD.